### ISSUE TITLE
Keep funcbench and prombench complitely separate.

### ISSUE THREAD

**Issue by krasi-georgiev** [prometheus/test-infra#356]
I think we need to keep funcbench and prombench completely separate with separate Makefiles. I know this will lead to few lines if duplicate code, but in the long run it would be easier to understand and maintain.

Since [gke is a package]("github.com/prometheus/test-infra/pkg/provider/gke") It will be simple enough to use it inside funcbench and extend it to support `funcbench gke resource ....`. 

Then funcbench will have its own docker image to use for the github actions - prominfra/funcbench. 

Then all [funcbench makefile targets](https://github.com/prometheus/test-infra/blob/master/prombench/Makefile#L49-L76) would be inside that docker image.

**Comment by nevill** [comment]
Yes, I know we discussed about it, but I may come up with new ideas. :)

IMO `funcbench` should not contain `mainifests` or `prombench` binary. Because
1. when `funcbench` is running inside k8s, it doesn't require `prombench` binary
2. `funcbench` will somehow be used by other projects, they may have their own infrastructure, or their own manifests.
3. `funcbench` doesn't require `prombench` when building.

> As we discussed, we will rename `prombench` to `infra`, I use `infra` instead hereafter.

In fact, all the manifests are parts of our test-infra, they should stay in the same container along with `infra`. Image `infra` are only used in GitHub actions. Container images `tools/*`, `funcbench` are used in kubernetes clusters. 

We can keep the `Makefile` separate by adjusting the directory layout.
for example, in the `infra` image:
```
funcbench/
  - Makefile
  - manifests
prombench/
  - Makefile
  - manifests
cluster-infra/
  - manifests
```

**Comment by geekodour** [comment]
If IUUC, this is the organization that you're suggesting:
```
├── ...
├── manifests/
│   ├── cluster-infra/
│   ├── README.md <- Contains setting up the Main Node
│   ├── cluster.yaml
│   ├── funcbench/ <-- Manifests and Makefile to deploy them with individual readme
│   └── prombench/ <-- Manifests and Makefile to deploy them with individual readme
├── ...
└── tools
    ├── ...
    ├── ...
    ├── funcbench <- /funcbench <- /tools/funcbench
    ├── infra <- this is the prombench cli
    └── ...
```

The `/tools/infra` docker image will have the `/manifests` inside it and have some kind of env var set which directory (`/`,`/manifests/funcbench`,`/manifests/prombench/` etc) should the `make` command be run from.

Is this right?

**Comment by nevill** [comment]
I think @geekodour you are talking about the project structure, while I was talking about docker image. But, the structure you proposed looks reasonable. 

Add a little modification.
```
├── manifests/
│   ├── cluster-infra/
│        ├── cluster.yaml
│        ├── k8s-resources/ <- old cluster-infra/* here
│   ├── README.md
│   ├── funcbench/
│   └── prombench/
├── pkg/
└── tools
    ├── prometheus-builder/ <- also moved here ?
    ├── ...
    └── ...
```


> The /tools/infra docker image will have the /manifests inside it and have some kind of env var set which directory (/,/manifests/funcbench,/manifests/prombench/ etc) should the make command be run from.

This is also good.

**Comment by geekodour** [comment]
fixed by #368
